### ISSUE TITLE
The --ipv6 flag to docker network create is broken for remote plugins in Docker 1.12.2

### ISSUE THREAD

**Issue by tomdee** [docker/docker#28055]
**Description**
If I run `docker network create` with a remote network driver and the --ipv6 flag, I expect my remote plugin to be passed the "V6" flag.

**Steps to reproduce the issue:**
1. Run dockerd with a cluster store configured - `sudo dockerd --cluster-store=etcd://127.0.0.1:2379`
2. Run etcd locally `docker run --detach --net=host quay.io/coreos/etcd etcd --advertise-client-urls "http://localhost:2379" --listen-client-urls "http://0.0.0.0:2379"`
3. Run the Calico Libnetwork plugin - `docker run --rm --net=host --privileged -e CALICO_ETCD_AUTHORITY=localhost:2379 -v /run/docker/plugins:/run/docker/plugins -v /var/run/docker.sock:/var/run/docker.sock calico/libnetwork-plugin:v1.0.0-beta-rc1 /libnetwork-plugin
4. Create a network -  `docker network create --ipv6=true  --driver=calico --ipam-driver=calico-ipam ipv6`

**Describe the results you received:**
Calico logs:
```
2016/11/04 01:30:34 GetCapabilities response JSON={"Scope":"global"}
2016/11/04 01:30:34 GetCapabilities response JSON={"RequiresMACAddress":false}
2016/11/04 01:30:34 GetDefaultAddressSpace response JSON={"LocalDefaultAddressSpace":"CalicoLocalAddressSpace","GlobalDefaultAddressSpace":"CalicoGlobalAddressSpace"}
2016/11/04 01:30:34 RequestPool JSON={"AddressSpace":"CalicoGlobalAddressSpace","Pool":"","SubPool":"","Options":{},"V6":false}
2016/11/04 01:30:34 RequestPool response JSON={"PoolID":"CalicoPoolIPv4","Pool":"0.0.0.0/0","Data":{"com.docker.network.gateway":"0.0.0.0/0"}}
2016-11-04 01:30:34.556470 I | Entering go-plugins-helpers createnetwork
2016/11/04 01:30:34 CreateNetwork JSON={"NetworkID":"20a4483125b4e7af19f370d12799ddfbb7ceb05dec4c9e89739db2b42135641b","Options":{"com.docker.network.enable_ipv6":true,"com.docker.network.generic":{}},"IPv4Data":[{"AddressSpace":"CalicoGlobalAddressSpace","Pool":"0.0.0.0/0","Gateway":"0.0.0.0/0","AuxAddresses":null}],"IPv6Data":[]}
2016/11/04 01:30:34 CreateNetwork response JSON={}
```

There are no logs from the Docker daemon

**Describe the results you expected:**
`"V6":false` should be `"V6":true` on the RequestPool

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
docker version
Client:
 Version:      1.12.2
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   bb80604
 Built:        Tue Oct 11 18:29:41 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.2
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   bb80604
 Built:        Tue Oct 11 18:29:41 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
docker info
Containers: 6
 Running: 2
 Paused: 0
 Stopped: 4
Images: 45
Server Version: 1.12.2
Storage Driver: overlay2
 Backing Filesystem: extfs
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: calico bridge null host overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: apparmor seccomp
Kernel Version: 4.8.2-040802-generic
Operating System: Ubuntu 16.04.1 LTS
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 31.25 GiB
Name: xeon
ID: CM6N:TGLQ:WIRF:5JGO:BX7D:ZPLC:GBBO:WEHD:U6C4:3PCD:WODP:T6FS
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Username: tomdee
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Cluster Store: etcd://localhost:2379
Insecure Registries:
 127.0.0.0/8

```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by cpuguy83** [comment]
ping @mavenugo

**Comment by aboch** [comment]
@tomdee Thanks for the report. It is not specific to network plugins. Same happen for inbuilt network drivers.

Libnetwork core does not conform to the auto pool allocation for v6.
For example, if `--ipv6` is specified but no IPv6 ``--subnet` is passed, libnetwork should still attempt a request to the IPAM driver for a v6 pool (besides the v4 pool) and fail the network creation if the IPAM driver can't return one.

I will take a look at it.

**Comment by tomdee** [comment]
Thanks @aboch

**Comment by tomdee** [comment]
Since https://github.com/docker/libnetwork/pull/1538 got merged, will this be fixed for 1.13?

**Comment by aboch** [comment]
Yes, it should.
